### ISSUE TITLE
EmptyFunctionBlock should not report interface default functions

### ISSUE THREAD

**Issue by Mygod** [arturbosch/detekt#1926]
## Expected Behavior of the rule

```
interface Stub {
    fun overrideMeOptionally() { }
}
```

## Context

This should not be reported.

detekt 1.0.1

**Comment by arturbosch** [comment]
Why not? The convention was to have a comment like pmd and sonarqube does.

**Comment by schalkms** [comment]
Thanks for reporting this. I agree, this case should be ignored by this rule.

**Comment by Mygod** [comment]
@arturbosch Like `fun overrideMeOptionally() { /* this is intentionally blank */ }`?

**Comment by marschwar** [comment]
Or rather `fun overrideMeOptionally() = Unit`

**Comment by Mygod** [comment]
Sure but that's less readable.

**Comment by cortinico** [comment]
> Why not? The convention was to have a comment like pmd and sonarqube does.

Agree here. I'm actually happier having a `// no-op` comment or similar to explain that the function is intentionally empty.

**Comment by Mygod** [comment]
That's ridiculous. If I want to say I haven't implemented a function, I can write `TODO()` in kotlin and that's the default behavior by IntelliJ's IDE as well.

**Comment by Mygod** [comment]
Furthermore I'm only talking about interface default blocks. If you ignore open funs and override funs then there should be an option to opt out these as well.

**Comment by arturbosch** [comment]
If the users want it, a config switch is fine by me. That's why I added the "Waiting for Feedback"-label.
However we should strive to prevail a good atmosphere in discussions.

**Comment by demonar** [comment]
> If the users want it, a config switch is fine by me. That's why I added the "Waiting for Feedback"-label.
> However we should strive to prevail a good atmosphere in discussions.

completely agree, hope my comment wasn't received as offensive, no intention at all.

**Comment by Mygod** [comment]
What other feedback are you looking for? I am only talking about interface default functions here. I am fine if you want a comment for final funs.
